src/main/java/uk/gov/dhsc/htbhf/controller/PostcodeController.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.dhsc.htbhf.controller;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.core.io.support.ResourcePatternResolver;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.PostConstruct;
+
+import static org.springframework.util.StringUtils.isEmpty;
+
+@Slf4j
+@Controller
+public class PostcodeController {
+
+    private static final Pattern POSTCODE_REGEX = Pattern.compile("^([A-Za-z]{2}[0-9]{1,2}[A-Za-z]{0,1}[0-9]{1,2}[A-Za-z]{2}).*");
+
+    private static final Map<String, ResponseEntity<String>> RESPONSES = readResponses();
```

Comment:
Might be worthwhile putting all the responses and the loading of them in their own file so the core controller logic (what there is of it anyway) clearly.

Reply:
I just had a quick go at it - there's almost nothing left in the controller then. Once the logic for reading files is removed (which means removing the responses themselves) the controller basically just becomes:
```
    private final Responses responses;

    @GetMapping(path = "/places/v1/addresses/postcode")
    public ResponseEntity<String> getAddressesForPostcode(@RequestParam("postcode") String postcode, @RequestParam(name = "key", required = false) String key) {
        return responses.getResponseForPostcode(postcode);
    }
```
Not sure it's worthwhile?

Reply:
I don't mind that to be honest, plus it means that any testing on the loading can be done without having to interact with the controller code.